Userland/Utilities/headless-browser.cpp

```diff
@@ -264,8 +277,216 @@ class HeadlessImageDecoderClient : public Web::ImageDecoding::Decoder {
     explicit HeadlessImageDecoderClient() = default;
 };
 
+static HashTable<RefPtr<Web::ResourceLoaderConnectorRequest>> s_all_requests;
+
 class HeadlessRequestServer : public Web::ResourceLoaderConnector {
 public:
+    class HTTPHeadlessRequest
+        : public Web::ResourceLoaderConnectorRequest
+        , public Weakable<HTTPHeadlessRequest> {
+    public:
+        static ErrorOr<NonnullRefPtr<HTTPHeadlessRequest>> create(String const& method, AK::URL const& url, HashMap<String, String> const& request_headers, ReadonlyBytes request_body)
+        {
+            auto stream_backing_buffer = TRY(ByteBuffer::create_uninitialized(1 * MiB));
+            auto underlying_socket = TRY(Core::Stream::TCPSocket::connect(url.host(), url.port().value_or(80)));
+            TRY(underlying_socket->set_blocking(false));
+            auto socket = TRY(Core::Stream::BufferedSocket<Core::Stream::TCPSocket>::create(move(underlying_socket)));
+
+            HTTP::HttpRequest request;
+            if (method.equals_ignoring_case("head"sv))
+                request.set_method(HTTP::HttpRequest::HEAD);
+            if (method.equals_ignoring_case("get"sv))
+                request.set_method(HTTP::HttpRequest::GET);
+            if (method.equals_ignoring_case("post"sv))
+                request.set_method(HTTP::HttpRequest::Invalid);
```

Comment:
So we just don't set the request method when the page requests an invalid or other (DELETE, PUT) request? Doesn't sound like a good idea to me.

Reply:
Thanks for the catch, I started writing that properly then got lost into the rabbit hole of "why is there only three options in HttpRequest" and never finished it! Fixed :)